gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcMembershipRepository.java

```diff
@@ -484,7 +484,14 @@ public Set<Membership> findByMemberIdAndMemberTypeAndReferenceTypeAndReferenceId
                         referenceId
                     );
             } else {
-                memberships = jdbcTemplate.query(query, getOrm().getRowMapper(), memberId, memberType.name(), referenceType.name());
+                memberships =
+                    jdbcTemplate.query(
+                        query + " and reference_id is null",
+                        getOrm().getRowMapper(),
+                        memberId,
+                        memberType.name(),
+                        referenceType.name()
+                    );
```

Comment:
Stupid question, could we get rid of the if(refId !=null) to have something like:
`memberships =
                    jdbcTemplate.query(
                        query + " and reference_id = ? or reference_id is null",
                        getOrm().getRowMapper(),
                        memberId,
                        memberType.name(),
                        referenceType.name(),
                        referenceId
                    );`
                    
to have only one query

Reply:
I don't think so. If you have two entries in you table one with the reference id value and another one with null you'll get both values but you want only one.